neqo-crypto/src/aead.rs

```diff
@@ -126,6 +126,39 @@ impl RealAead {
         Ok(&output[0..(l.try_into()?)])
     }
 
+    /// Encrypt `data` consisting of `aad` and plaintext `input` in place.
+    ///
+    /// The space provided in `data` needs to allow `Aead::expansion` more bytes to be appended.
+    ///
+    /// # Errors
+    ///
+    /// If the input can't be protected or any input is too large for NSS.
+    pub fn encrypt_in_place(
+        &self,
+        count: u64,
+        aad: Range<usize>,
```

Comment:
I think that it would be better to take a slice for the AAD.  Then it could be kept elsewhere, if that makes more sense.

Reply:
I'm not quite sure how to do this without copying `aad`, since there'd then be two references into `data`?

Reply:
You will have a `&mut [u8]` that can be split (in this case temporarily) into an AAD and plaintext or ciphertext part. See https://doc.rust-lang.org/stable/std/primitive.slice.html#method.split_at_mut